### PR TITLE
[BottomSheet] Safe Area fixes

### LINKED ISSUE #2005: [BottomSheet] Support Safe Area
### Overview
Issues visible in the Catalog: Full screen is unusable because of the top notch. Potential issues with the Home indicator and the content. On portrait, the content doesn’t adhere to the safe area.

bottomsheetlandscape

bottomsheetfullscreen

### Version number
35.0.0

### Operating system and device
iOS 11 / iPhoneX

### PR DESCRIPTION
**Pull request by chuga**
- Adjusts the max height to prevent the content from going below the top safe area inset.
- Sets the scroll view's bottom inset to at least  the bottom safe area inset.
- Stops ignoring the contentInset when calculating the content size of the scroll view.
- Adjusts preferredSheetHeight to include the bottom safe area inset when used internally.

Closes #2005

### CODE REVIEW

**Review by samnm — APPROVED** [review]
🎉💯🔥

**Comment by chuga** [comment]
🙌

File: components/BottomSheet/src/private/MDCSheetContainerView.m [hunk 1]
```diff
@@ -215,14 +249,20 @@ - (CGFloat)truncatedPreferredSheetHeight {
 
 // Returns the maximum allowable height that the sheet can be dragged to.
 - (CGFloat)maximumSheetHeight {
-  CGSize size = self.bounds.size;
-  CGFloat scrollViewContentHeight = self.sheet.scrollView.contentSize.height;
+  CGFloat boundsHeight = self.bounds.size.height;
```
Comment (ianegordon) [inline components/BottomSheet/src/private/MDCSheetContainerView.m]: CGRectGetHeight() since bounds may not be standardized.

**Review by ianegordon — CHANGES_REQUESTED** [review]

File: components/BottomSheet/src/private/MDCSheetContainerView.m (hunk 1 above)
Comment (chuga) [reply components/BottomSheet/src/private/MDCSheetContainerView.m]: Done (we were already using self.bounds.size.height fwiw).
PTAL

File: components/BottomSheet/src/private/MDCSheetContainerView.m [hunk 2]
```diff
@@ -137,6 +145,25 @@ - (void)traitCollectionDidChange:(UITraitCollection *)previousTraitCollection {
...
+  if (@available(iOS 11.0, *)) {
+    [super safeAreaInsetsDidChange];
+
+    _preferredSheetHeight = self.originalPreferredSheetHeight + self.safeAreaInsets.bottom;
+
+    UIEdgeInsets contentInset = self.sheet.scrollView.contentInset;
+    contentInset.bottom = MAX(contentInset.bottom, self.safeAreaInsets.bottom);
+    self.sheet.scrollView.contentInset = contentInset;
+
+    CGRect scrollViewFrame = self.sheet.scrollView.frame;
```
Comment (ianegordon) [inline components/BottomSheet/src/private/MDCSheetContainerView.m]: Standardize before accessing the struct directly.
CGRect scrollViewFrame = CGRectStandardize(self.sheet.scrollView.frame);

**Review by ianegordon — CHANGES_REQUESTED** [review]

File: components/BottomSheet/src/private/MDCSheetContainerView.m (hunk 2 above)
Comment (chuga) [reply components/BottomSheet/src/private/MDCSheetContainerView.m]: What's the reasoning behind that?

Comment (ianegordon) [reply components/BottomSheet/src/private/MDCSheetContainerView.m]: We are directly accessing the struct directly below.
To ensure we are getting valid value out of a bounds / frame, we can either use the accessor functions like CGRectGetWidth() / Height() / etc. -OR- we can CGRectStandardize() the frame and access the standardized struct values directly.

Note that Swift eliminates all of this by only returning standardized values.  Our bright, safe Swifty future awaits! 😄

File: components/BottomSheet/src/private/MDCSheetContainerView.m (hunk 1 above)
Comment (ianegordon) [reply components/BottomSheet/src/private/MDCSheetContainerView.m]: Indeed, we definitely haven't been as strict as we should have been in the past, but if we are touching the code.....

**Comment by chuga** [comment]
RIP my place in the Travis queue 😔
PTAL

**Review by ianegordon — APPROVED** [review]

**Pull request merged**